### ISSUE TITLE
Neovim when called through $EDITOR is broken

### ISSUE THREAD

**Issue by Granitosaurus** [xonsh/xonsh#3672]
When opening `neovim` through a program that subprocess `$EDITOR` the editor is unusable in xonsh; e.g. `git commit`. 
Here's a screenshot:
![image](https://user-images.githubusercontent.com/5476164/89758423-447ba880-dad7-11ea-96a4-332fb3e08642.png)
In the example above I called command `kubectl edit` which opens `$EDITOR` with temporary file. This works absolutely fine with other editors such as `vim` or calling `nvim` in `bash` but something between `xonsh` and `neovim` is not working.

## xonfig

<details>

```
+------------------+----------------------+
| xonsh            | 0.9.19.dev8          |
| Git SHA          | 0fe76b53             |
| Commit Date      | Aug 10 06:28:52 2020 |
| Python           | 3.8.5                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 3.0.5                |
| shell type       | prompt_toolkit       |
| pygments         | 2.6.1                |
| on posix         | True                 |
| on linux         | True                 |
| distro           | arch                 |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

</details>

## Expected Behavior

Should open `neovim` and edit file the same way opening `nvim` process would

## Steps to Reproduce

1. Install `neovim` (tested with v0.4.4)  
2. set is as EDITOR: `$EDITOR='nvim'`  
3. call edit command like `git commit`

**Comment by scopatz** [comment]
Hmm that is odd. I am not seeing this with `xo`, my text editor.  Thanks for reporting @Granitosaurus!

**Comment by daniel-shimon** [comment]
Can confirm on WSL2

**Comment by dyuri** [comment]
Can confirm it myself (Linux). Did a quick check, in case of `git commit` `get_predictor_threadable` gets `git` (obviously) and git itself is threadable. It's not called when `nvim` is executed later.
Using `$THREAD_SUBPROCS=False` or `$[git commit]` works.

**Comment by scopatz** [comment]
Yeah, I think if you are in this situation, you need to set `git` or `git commit` to be not threadable.  I don't know why this is only cropping up with `nvim` though

**Comment by Granitosaurus** [comment]
@scopatz can you set some programs as non-threadable in xonsh? 
I've experienced the same issue with `ranger` recently when a program called it through mimetype open (i.e. xdg-open) but I'm not sure how to replicate it right now.

**Comment by scopatz** [comment]
Yes, you can always do `__xonsh__.commands_cache.threadable_predictors["my-cmd"] = lambda *a, **kw: False`

**Comment by scopatz** [comment]
We probably should have a command for that

**Comment by lazmol** [comment]
I am also affected by this. But not just through $EDITOR, running xonsh -c nvim from a xonsh prompt gives me the same problem as screenshoted above. I have set shell=/bin/bash in my neovim config.
What do I have to do to make it non-threadable then?
Add these to my .xonshrc:
`__xonsh__.commands_cache.threadable_predictors["nvim"] = lambda *a, **kw: False`
`__xonsh__.commands_cache.threadable_predictors["git"] = lambda *a, **kw: False`
?

**Comment by dyuri** [comment]
`nvim` should be already flagged as non-threadable, so setting it non-threadable again won't change anything. Maybe it is ignored in case of single command mode?

**Comment by Mikadore** [comment]
Hi. This issue still persists for me. I'm on Linux, and `git commit` is broken with vi/vim/nvim

**Comment by daniel-shimon** [comment]
Hi @Mikadore!

1. Are you using the latest version of xonsh `0.10.1`?
2. Is there some alias defined for `git` in your setup?
3. In what way do those commands break for you?

**Comment by Mikadore** [comment]
> Hi @Mikadore!
> 
> 1. Are you using the latest version of xonsh `0.10.1`?
> 2. Is there some alias defined for `git` in your setup?
> 3. In what way do those commands break for you?

Hey!

1. No, `0.10.1`  is broken for me (should probably open an issue :D), so I'm stuck at `0.9.27` 
2. No
3.  'Special' keys don't work, that is the arrow keys, escape, and control + X are broken as shown below. Also, pressing `ctrl + c` just kills nvim.

![2021-08-22 16-34-29](https://user-images.githubusercontent.com/35429782/130359466-c1464946-358e-40ac-a282-19680ba67e8d.gif)


<details>
<summary> xonfig </summary>

```
+------------------+----------------------+
| xonsh            | 0.9.27               |
| Git SHA          | 71fe9014             |
| Commit Date      | Jan 29 08:58:58 2021 |
| Python           | 3.9.6                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 3.0.19               |
| shell type       | prompt_toolkit       |
| pygments         | 2.10.0               |
| on posix         | True                 |
| on linux         | True                 |
| distro           | arch                 |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
| on jupyter       | False                |
| jupyter kernel   | None                 |
| xontrib          | []                   |
+------------------+----------------------+
```
</details>

**Comment by daniel-shimon** [comment]
Oh OK so the fix was merged to `0.10.0`, so it makes sense this doesn't work for you :)

**Comment by Mikadore** [comment]
Oops, my my bad :sweat_smile:
